### PR TITLE
Fix unsafe component will mount

### PR DESCRIPTION
**Pull request by rianfowler**
- recent react code mod introduced eslint errors into master (make sure to pull down master and branch from latest- i don't think our build server is merging master before doing the linting checks)
- [camelcase](https://eslint.org/docs/rules/camelcase) supports an allowed list this but since i couldn't tell what version of eslint received this update, I just updated to the latest eslint which required updating to the latest airbnb
- i ignored all of the new airbnb rules that caused errors

### CODE REVIEW

File: packages/vagov-eslint/.eslintrc [hunk 1]
```diff
@@ -27,7 +27,7 @@
   },
   "rules": {
     // Override airbnb style.
-    "camelcase": [2, { "properties": "always" }],
+    "camelcase": [2, {"allow": ["UNSAFE_componentWillMount"]}],
```
Comment (rianfowler) [inline packages/vagov-eslint/.eslintrc]: new camel case rule

File: packages/vagov-eslint/package.json [hunk 2]
```diff
@@ -6,8 +6,8 @@
         "vagov-eslint": "./bin/vagov-eslint"
     },
     "dependencies": {
-        "eslint": "4.4.1",
-        "eslint-config-airbnb": "^15.1.0",
+        "eslint": "5.15.3",
+        "eslint-config-airbnb": "^17.1.0",
```
Comment (rianfowler) [inline packages/vagov-eslint/package.json]: new eslint and airbnb

File: packages/vagov-eslint/.eslintrc [hunk 3]
```diff
@@ -81,30 +81,46 @@
...
       { "callbacksLast": true, "shorthandFirst": true }
     ], // TODO(awong): Too hard to turn on.
+    "react/jsx-wrap-multilines": 0,
+    "react/no-access-state-in-setstate": 0,
     "react/no-array-index-key": 0,
     "react/no-unescaped-entities": 0,
     "react/no-unused-prop-types": 0,
-    "react/require-default-props": 0
+    "react/require-default-props": 0,
+    "react/sort-comp": 0
```
Comment (rianfowler) [inline packages/vagov-eslint/.eslintrc]: some new ignored stuff

File: packages/vagov-eslint/.eslintrc [hunk 4]
```diff
@@ -81,30 +81,46 @@
...
     "import/extensions": 0,
     "import/first": 0,
     "import/no-named-as-default": 0,
+    "import/order": 0,
     "import/prefer-default-export": 0,
+    "jsx-a11y/anchor-is-valid": 0,
+    "jsx-a11y/click-events-have-key-events": 0,
     "jsx-a11y/href-no-hash": 0,
+    "jsx-a11y/label-has-associated-control": 0,
+    "jsx-a11y/label-has-for": 0,
```
Comment (jbalboni) [inline packages/vagov-eslint/.eslintrc]: Some of these sound like they'd be useful to have one. Maybe we can make an issue to evaluate them?

Comment (rianfowler) [reply packages/vagov-eslint/.eslintrc]: sounds good- i'll make a ticket for that

**Pull request merged**